app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java

```diff
@@ -276,4 +285,15 @@ private void fetchHomeDetails() {
             new Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS)));
     dialog.setCancelable(false).show();
   }
+
+  @Override public boolean onQueryTextSubmit(String query) {
+    presenter.searchLocation(query);
+    searchMenuItem.collapseActionView();
+    //return true to indicate that it has handled the submit request
```

Comment:
comment not required. Is collapsing the action view desired. The search term might be lost.